common/viperutil/config_util.go

```diff
@@ -202,6 +203,59 @@ func getKeysRecursively(base string, getenv envGetter, nodeKeys map[string]inter
 	return result
 }
 
+type viperGetter func(key string) interface{}
+
+func getViperKeysRecursively(base string, getKey viperGetter, nodeKeys map[string]interface{}, oType reflect.Type) map[string]interface{} {
```

Comment:
Was this the function that added so much runtime to our tests? Have you profiled the runtime before/after this change?

Reply:
On first glance, it does seem like the overall time to complete UT increases significantly with this code... 27 minutes v. 12-13 minutes for the other PRs out there. Force-pushed to this PR to retrigger but I'll see if there's anything else we can do to fix these overrides without reviving this inefficient code.

Reply:
Attempting to incorporate the same caching mechanism that was added to improve the performance for the orderer/configtxgen. We'll see if that ends up with the same performance boost.

Reply:
Abandoned this horribly inefficient viper usage. :)

Reply:
Actually, if you look at the CI analytics, it seems the jump in total UT time is unrelated: https://dev.azure.com/Hyperledger/Fabric/_pipeline/analytics/duration?definitionId=50&contextType=build